### DISCUSSION TITLE
Allow IEnumerable instead of HashSet in server broadcasting + improve performance in SceneManager

### DISCUSSION THREAD

**Discussion by ghost** [FirstGearGames/FishNet#433] [Feature Request]
In `ServerManager.Broadcast(connections, ...);` instead of `HashSet<NetworkConnection>` use `IEnumerable<NetworkConnection>` since it's more general & does not break anything.

Plus, in SceneManager, like 1779, you literally allocate an array just to iterate through it. In `RebuildObservers(...)` method also allow `IEnumerable<NetworkObject>` and in line mentioned above remove `.ToArray()` call.

**Comment by FirstGearGames** [comment]
Edited:
We are moving away from IEnumerable except in areas where enumerator allocations are unavoidable. Method signatures will be replaced with IList.

I can however optimize the SceneManager methods and a few more in there.

**Comment by FirstGearGames** [comment]
Improved where applicable. Available in 3.10.8R
